full tutorial review pre-trial 02 (#34)
epiverse-trace/tutorials-middle: episodes/severity-static.Rmd

---

> title: 'Estimate severity'

Reviewer: Estimation of  outbreak severity

![HHS Pandemic Planning Scenarios based on the Pandemic Severity Assessment Framework. This uses a combined measure of clinical severity and transmissibility to characterise influenza pandemic scenarios. HHS: United States Department of Health and Human Services ([CDC, 2016](https://www.cdc.gov/flu/pandemic-resources/national-strategy/severity-assessment-framework-508.html)).](fig/cfr-hhs-scenarios-psaf.png){alt='The horizontal axis is the scaled measure of clinical severity, ranging from 1 to 7, where 1 is low, 4 is moderate, and 7 is very severe. The vertical axis is the scaled measure of transmissibility, ranging from 1 to 5, where 1 is low, 3 is moderate, and 5 is highly transmissible. On the graph, HHS pandemic planning scenarios are labeled across four quadrants (A, B, C and D). From left to right, the scenarios are “seasonal range,” “moderate pandemic,” “severe pandemic” and “very severe pandemic.” As clinical severity increases along the horizontal axis, or as transmissibility increases along the vertical axis, the severity of the pandemic planning scenario also increases.'}

> One epidemiological approach to estimating the clinical severity is quantifying the Case Fatality Risk or CFR. CFR is the conditional probability of death given confirmed diagnosis, calculated as the cumulative number deaths from an infectious disease over the number of confirmed diagnosed cases. However, calculating this directly during the course of an epidemic tends to result in a naive or biased CFR given the time [delay](../learners/reference.md#delaydist) from onset to death, varying substantially as the epidemic progresses and stabilising at the later stages of the outbreak ([Ghani et al., 2005](https://academic.oup.com/aje/article/162/5/479/82647?login=false#620743)).

Reviewer: (CFR)

> One epidemiological approach to estimating the clinical severity is quantifying the Case Fatality Risk or CFR. CFR is the conditional probability of death given confirmed diagnosis, calculated as the cumulative number deaths from an infectious disease over the number of confirmed diagnosed cases. However, calculating this directly during the course of an epidemic tends to result in a naive or biased CFR given the time [delay](../learners/reference.md#delaydist) from onset to death, varying substantially as the epidemic progresses and stabilising at the later stages of the outbreak ([Ghani et al., 2005](https://academic.oup.com/aje/article/162/5/479/82647?login=false#620743)).

Reviewer: of deaths from...

:::::::::::::::::::::::

More generally, estimating severity can be helpful even outside of a pandemic planning scenario and in the context of routine public health. Knowing whether an outbreak has or had a different severity from the historical record can motivate causal investigations, which could be intrinsic to the infectious agent (e.g. a new, more severe strain) or due to underlying factors in the population (e.g. reduced immunity or morbidity factors) ([Lipsitch et al., 2015](https://journals.plos.org/plosntds/article?id=10.1371/journal.pntd.0003846)).

In this tutorial we are going to learn how to use the `{cfr}` package to calculate and adjust a CFR estimation using [delay distributions](../learners/reference.md#delaydist) from `{epiparameter}` or elsewhere, based on the methods developed by [Nishiura et al., 2009](https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0006852), also, how we can reuse `{cfr}` functions for more severity measurements.

We’ll use the pipe `%>%` to connect some of their functions, so let’s also call to the `{tidyverse}` package:

```{r,message=FALSE,warning=FALSE}

> library(cfr)

Reviewer 2: ```suggestion
library(cfr)
library(covidregionaldata)
```

Author: thanks, we'll omit covidregionaldata as reported in #36

To calculate the naive CFR, the `{cfr}` package requires an input data frame with three columns named:

- `date`
- `cases`
- `deaths`

Let's explore the `ebola1976` dataset, which comes from the first Ebola outbreak in what was then called Zaire (now the Democratic Republic of the Congo) in 1976, as analysed by Camacho et al. (2014).

```{r}
# Load the Ebola 1976 data provided with the package
data("ebola1976")

# view top data
ebola1976 %>% as_tibble()
```

:::::::::::::::::: callout

### We need aggregated incidence data

`{cfr}` reads **aggregated** incidence data. Similar to the `{EpiNow2}` with the difference that for `{cfr}` we need one more column named `deaths`.

```{r}
EpiNow2::example_confirmed %>% as_tibble()
```

This data input should be **aggregated** by day, which means one observation *per day*, containing the *daily* number of reported cases and deaths. Observations with zero or missing values should also be included, similar to a time-series data.

Also, `{cfr}` currently works for *daily* data only, not weekly data.

<!-- suggest ways to deal with raw input weekly data -->
<!-- https://github.com/epiverse-trace/cfr/issues/117 -->

> ### Aggregated differs from Linelist

Reviewer: Here you can make the link with the relevant early tutorial sections.

> Lastly, *aggregated* incidence data differs from **linelist** data, where each observation contains subject-level data.

Reviewer: subject-level information

[Fraser et al., 2009](https://www.science.org/doi/full/10.1126/science.1176062) reinterpreted the data assessing the biases and getting a clinical severity lower than the 1918 influenza pandemic but comparable with that seen in the 1957 pandemic.

::::::::::::::::::::

:::::::::::::::::::: discussion

Based on your experience:

- Share any previous outbreak in which you participated in its response.

Answer to these questions:

- How did you assess the clinical severity of the outbreak?
- What were the primary sources of bias?
- What did you do to take into account the identified bias?
- What complementary analysis would you do to solve the bias?

:::::::::::::::::::: 

## Delay-adjusted CFR

[Nishiura et al., 2009](https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0006852) developed a method that considers the **time delay** from the onset of symptoms to death.

> Real-time outbreaks may have a number of deaths that are insufficient to determine the time distribution between onset and death. Therefore, we can estimate the _distribution delay_ from historical outbreaks or reuse the ones accessible via R packages like `{epiparameter}` or `{epireview}`, which collects them from literature searches and systematic reviews.

Reviewer 2: ```suggestion
Real-time outbreaks may have a number of deaths that are insufficient to determine the time distribution between onset and death. Therefore, we can estimate the _distribution delay_ from historical outbreaks or reuse the ones accessible via R packages like `{epiparameter}` or `{epireview}`, which collect them from published scientific literature.
```

```{r}
# Calculate the static CFR while adjusting for delays
cfr_static(
  data = ebola1976,
  delay_density = function(x) density(onset_to_death_ebola, x)
)
```

```{r,echo=FALSE}
out_delay_adjusted <-
  cfr_static(
    data = ebola1976,
    delay_density = function(x) density(onset_to_death_ebola, x)
  )

out_mean <- out_delay_adjusted %>% pull(severity_mean)
out_low <- out_delay_adjusted %>% pull(severity_low)
out_high <- out_delay_adjusted %>% pull(severity_high)
```

The delay-adjusted CFR indicated that the overall disease severity _at the end of the outbreak_ or with the _latest data available at the moment_ is `r out_mean` with a 95% confidence interval between `r out_low` and `r out_high`, slightly higher than the naive one.

::::::::::::::::::::::::::: spoiler

### How does {cfr} works?

To adjust the CFR, [Nishiura et al., 2009](https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0006852) use the case and death incidence data to estimate the number of cases with known outcomes:

$$
  u_t = \dfrac{\sum_{i = 0}^t
        \sum_{j = 0}^\infty c_{i - j} f_{j}}{\sum_{i = 0} c_i},
$$

where:

> - $c_{t}$ is the daily case incidence at time $t$, 

Reviewer 2: In the rendered version the font here is really big compared to the rest of the text

Author: is a feature of the workbench. we should fill an issue there to understand the rationale or how to edit locally in our branch

$u_{t}$ is used to **scale** the value of the cumulative number of cases in the denominator in the calculation of the CFR. This is calculated internally with the [`estimate_outcomes()`](https://epiverse-trace.github.io/cfr/reference/estimate_outcomes.html) function.

The estimator for CFR can be written as: 

$$p_{t} = \frac{b_{t}}{u_{t}}$$

where $p_{t}$ is the realized proportion of confirmed cases to die from the infection (or the unbiased CFR), and $b_{t}$, the crude and biased estimate of CFR (also naive CFR).

From this last equation, we observe that the unbiased CFR $p_{t}$ is larger than biased CFR $b_{t}$ because in $u_{t}$ the numerator is smaller than the denominator (note that $f_{t}$ is the probability distribution of the *delay distribution* between onset and death). Therefore, we refer to $b_{t}$ as the biased estimator of CFR.

When we observe the entire course of an epidemic (from $t \rightarrow \infty$), $u_{t}$ tends to 1, making $b_{t}$ tends to $p_{t}$ and become an unbiased estimator ([Nishiura et al., 2009](https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0006852)).

:::::::::::::::::::::::::::

:::::::::::::::::: callout

> ### Use the `<epidist>` class

Reviewer 2: here too the word epidist is very small and there's no spaces between words

Author: will fix #41

When using an `<epidist>` class object we can use this expression as a template:

`function(x) density(<EPIDIST_OBJECT>, x)`

For distribution functions with parameters not available in `{epiparameter}`, we suggest you two alternatives: 

- Create an `<epidist>` class object, to plug into other R packages of the outbreak analytics pipeline. Read the [reference documentation of `epiparameter::epidist()`](https://epiverse-trace.github.io/epiparameter/reference/epidist.html), or

- Read `{cfr}` vignette for [a primer on working with delay distributions](https://epiverse-trace.github.io/cfr/articles/delay_distributions.html).

::::::::::::::::::

:::::::::::::::::: spoiler

## When to use discrete distributions?

For  `cfr_static()` and all the `cfr_*()` family of functions, the most appropriate choice to pass are **discrete** distributions. This is because we will work with daily case and death data.

We can assume that evaluating the Probability Distribution Function (PDF) of a *continuous* distribution is equivalent to the Probability Mass Function (PMF) of the equivalent *discrete* distribution.

> However, this assumption may not be appropriate for distributions with greater picks. For example, when onset-to-death is strongly peaked with a low variance. In such cases, the average disparity between the PDF and PMF is expected to be more pronounced compared to distributions with broader spreads.

Reviewer 2: ```suggestion
However, this assumption may not be appropriate for distributions with larger peaks. For instance, diseases with an onset-to-death distribution that is strongly peaked with a low variance. In such cases, the average disparity between the PDF and PMF is expected to be more pronounced compared to distributions with broader spreads.
```

> In a scenario like the last one, you can opt to discretise the continuous distribution using `epiparameter::discretise()` to an `<epidist>` object.

Reviewer 2: which scenario? needs clarification

::::::::::::::::::

:::::::::::::::::::::::::::::::::::::::: challenge

Load the `ebola_kikwit_1995` from `{outbreaks}` and estimate the delay-adjusted CFR using the appropriate distribution delay. Then:

- Compare the naive and the delay-adjusted CFR solutions!

:::::::::::::::::::: hint

> - Find the appropriate `<epidist>` object!

Reviewer 2: what does this mean? in the solution you're using the same epidist object that was used for ebola1976

Author: just realized this! I think I assumed it was a different disease. I'll check if any other fits better. #42

::::::::::::::::::::

:::::::::::::::::::: solution

```{r}
library(outbreaks)

outbreaks::ebola_kikwit_1995 %>% glimpse()

outbreaks::ebola_kikwit_1995 %>%
  rename(
    cases = onset,
    deaths = death
  ) %>%
  cfr::cfr_static(
    delay_density = function(x) density(onset_to_death_ebola, x)
  )
```

::::::::::::::::::::

::::::::::::::::::::::::::::::::::::::::

## An early-stage CFR estimate

> We just found that the difference between the _naive_ and _delay-adjusted_ CFR estimates using `cfr_static()` are _just slightly_ different. So, what is the main difference?

Reviewer 2: ```suggestion
On the challenge above, we discovered that the difference between the _naive_ and _delay-adjusted_ CFR estimates using `cfr_static()` are _just slightly_ different. So, what is the main difference?
```

> `{cfr}` method aims to obtain an unbiased estimator "well before" observing the entire course of the outbreak. For this `{cfr}` uses the underestimation factor $u_{t}$ to estimate the unbiased CFR $p_{t}$ using maximum-likelihood methods, given the sampling process defined by [Nishiura et al., 2009](https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0006852).

Reviewer 2: ```suggestion
`{cfr}` method aims to obtain an unbiased estimator "well before" observing the entire course of the outbreak. For this, `{cfr}` uses the underestimation factor $u_{t}$ to estimate the unbiased CFR $p_{t}$ using maximum-likelihood methods, given the sampling process defined by [Nishiura et al., 2009](https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0006852).
```

From *aggregated incidence data*, at time $t$ we know the cumulative number of confirmed cases and deaths, $C_{t}$ and $D_{t}$, and wish to estimate the unbiased CFR $\pi$, by way of the factor of underestimation $u_{t}$. 

If we knew the factor of underestimation $u_{t}$ we could specify the size of the population of confirmed cases no longer at risk ($u_{t}C_{t}$, **shaded**), although we do not know which surviving individuals belong to this group. A proportion $\pi$ of those in the group of cases still at risk (size $(1- u_{t})C_{t}$, **unshaded**) is expected to die.

Because each case no longer at risk had an independent probability of dying, $\pi$, the number of deaths, $D_{t}$, is a sample from a binomial distribution with sample size $u_{t}C_{t}$, and probability of dying $p_{t}$ = $\pi$.

This is represented by the following likelihood function to obtain the maximum likelihood estimate of the unbiased CFR $p_{t}$ = $\pi$:

$$
  {\sf L}(\pi | C_{t},D_{t},u_{t}) = \log{\dbinom{u_{t}C_{t}}{D_{t}}} + D_{t} \log{\pi} +
  (u_{t}C_{t} - D_{t})\log{(1 - \pi)},
$$

This estimation is performed by the internal function `?cfr:::estimate_severity()`.

::::::::::::::::::::::::::

> The `cfr_static()` estimate is useful to get an overall severity estimate of the outbreak (so far). Once the outbreak has clearly ended, or has progressed such that deaths are reported more, the estimated CFR is then _closest to_ the 'true' unbiased CFR.

Reviewer 2: "or has progressed such that deaths are reported more" <- what does this mean?

Author: "Once the outbreak has ended or has progressed such that more deaths are reported, the estimated CFR is then _closest to_ the 'true' unbiased CFR."

:::::::::::::::::::::::::::::::::::::::::::: discussion

### Interpret the early-stage CFR estimate

Based on the figure above:

- How much difference in days is between the date in which the 95% CI of the estimated _delay-adjusted CFR_ vs _naive CFR_ cross with the CFR estimated at the end of the outbreak?

Discuss:

- What are the Public health policy implications of having a _delay-adjusted CFR_ estimate?

:::::::::::::::::::::::::::::::::::::::::::: 

:::::::::::::::::::::: hint

We can use either visual inspection or analysis of the output data frames.

::::::::::::::::::::::

:::::::::::::::::::::: solution

There is almost one month of difference.

Note that the estimate has considerable uncertainty at the beginning of the time series. After two weeks, the delay-adjusted CFR approaches the overall CFR estimate at the outbreak's end.

Is this pattern similar to other outbreaks? We can use the data sets in this episode's challenges. We invite you to find it out!

::::::::::::::::::::::

::::::::::::::::: spoiler

### rolling or time-varying?

> The `{cfr}` package has a function called `cfr_time_varying()` with different functionality that makes them not comparable.

Reviewer 2: ```suggestion
The `{cfr}` package has a function called `cfr_time_varying()` with functionality that differs from `cfr_static()`.
```
